pkg/go/graph/weighted_graph.go

```diff
@@ -0,0 +1,409 @@
+package graph
+
+import (
+	"fmt"
+	"math"
+	"strings"
+)
+
+const Infinite = math.MaxInt32
+
+var errModelCycle = fmt.Errorf("model cycle")
```

Comment:
error constants should be exported from the package by capitalizing the first character of the name.

by exporting the errors, dependent packages can call `errors.Is(err, graph.ErrModelCycle)` on any encountered error.

since there is no formatting, you can use `errors.New()` to create the errors.

Reply:
noted, making the changes